### PR TITLE
[13.0][MIG] web_company_color: Migration to 13.0

### PR DESCRIPTION
**Pull request by Tardo**
**Superseded** https://github.com/OCA/web/pull/1552

Well, since I'm the author of the module and the changes for 13.0 are not trivial... i have done it to learn a bit more about Odoo :P

### CODE REVIEW

**Comment by LoisRForgeFlow** [comment]
@Tardo Thanks a lot!!

I am testing locally and I got the following error while installing it:

<details>

```
2020-04-07 12:23:54,744 24502 INFO mrp_production_request_v13 odoo.modules.loading: Module base_sparse_field: loading demo 
2020-04-07 12:23:54,983 24502 INFO mrp_production_request_v13 odoo.modules.registry: module web_company_color: creating or updating database tables 
2020-04-07 12:23:55,032 24502 INFO mrp_production_request_v13 odoo.modules.loading: loading web_company_color/view/assets.xml 
2020-04-07 12:23:55,048 24502 INFO mrp_production_request_v13 odoo.modules.loading: loading web_company_color/view/res_company.xml 
2020-04-07 12:23:55,077 24502 INFO mrp_production_request_v13 odoo.modules.loading: Module web_company_color: loading demo 
2020-04-07 12:23:55,097 24502 INFO mrp_production_request_v13 odoo.modules.loading: 29 modules loaded in 0.92s, 0 queries 
2020-04-07 12:23:55,395 24502 INFO mrp_production_request_v13 odoo.modules.loading: Modules loaded. 
2020-04-07 12:23:55,402 24502 INFO mrp_production_request_v13 odoo.addons.base.models.ir_module: getting next ir.actions.todo() 
2020-04-07 12:23:55,405 24502 INFO mrp_production_request_v13 odoo.modules.registry: Registry changed, signaling through the database 
2020-04-07 12:23:55,406 24502 INFO mrp_production_request_v13 werkzeug: 127.0.0.1 - - [07/Apr/2020 12:23:55] "POST /web/dataset/call_button HTTP/1.1" 200 - 1684 0.574 1.475
2020-04-07 12:23:55,419 24502 INFO mrp_production_request_v13 odoo.addons.base.models.ir_http: Generating routing map for key None 
2020-04-07 12:23:56,585 24502 INFO mrp_production_request_v13 odoo.models.unlink: User #2 deleted ir.attachment records with IDs: [664] 
2020-04-07 12:23:57,399 24502 INFO mrp_production_request_v13 odoo.models.unlink: User #2 deleted ir.attachment records with IDs: [666] 
2020-04-07 12:23:57,438 24502 INFO mrp_production_request_v13 werkzeug: 127.0.0.1 - - [07/Apr/2020 12:23:57] "GET /web HTTP/1.1" 500 - 1056 0.469 1.552
2020-04-07 12:23:57,445 24502 ERROR mrp_production_request_v13 werkzeug: Error on request:
Traceback (most recent call last):
  File "/home/lois/odoo-env/odoo13/lib/python3.6/site-packages/werkzeug/serving.py", line 270, in run_wsgi
    execute(self.server.app)
  File "/home/lois/odoo-env/odoo13/lib/python3.6/site-packages/werkzeug/serving.py", line 258, in execute
    application_iter = app(environ, start_response)
  File "/home/lois/odoo/13.0/odoo/odoo/service/server.py", line 439, in app
    return self.app(e, s)
  File "/home/lois/odoo/13.0/odoo/odoo/service/wsgi_server.py", line 142, in application
    return application_unproxied(environ, start_response)
  File "/home/lois/odoo/13.0/odoo/odoo/service/wsgi_server.py", line 117, in application_unproxied
    result = odoo.http.root(environ, start_response)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 1287, in __call__
    return self.dispatch(environ, start_response)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 1257, in __call__
    return self.app(environ, start_wrapped)
  File "/home/lois/odoo-env/odoo13/lib/python3.6/site-packages/werkzeug/wsgi.py", line 766, in __call__
    return self.app(environ, start_response)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 1456, in dispatch
    result = ir_http._dispatch()
  File "/home/lois/odoo/13.0/odoo/addons/auth_signup/models/ir_http.py", line 19, in _dispatch
    return super(Http, cls)._dispatch()
  File "/home/lois/odoo/13.0/odoo/addons/web_editor/models/ir_http.py", line 21, in _dispatch
    return super(IrHttp, cls)._dispatch()
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_http.py", line 233, in _dispatch
    return cls._handle_exception(e)
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_http.py", line 201, in _handle_exception
    return request._handle_exception(exception)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 750, in _handle_exception
    return super(HttpRequest, self)._handle_exception(exception)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/lois/odoo/13.0/odoo/odoo/tools/pycompat.py", line 14, in reraise
    raise value
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_http.py", line 229, in _dispatch
    result = request.dispatch()
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 809, in dispatch
    r = self._call_function(**self.params)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 350, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/lois/odoo/13.0/odoo/odoo/service/model.py", line 93, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 342, in checked_call
    result.flatten()
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 1236, in flatten
    self.response.append(self.render())
  File "/home/lois/odoo/13.0/odoo/odoo/http.py", line 1229, in render
    return env["ir.ui.view"].render_template(self.template, self.qcontext)
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_ui_view.py", line 1174, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/home/lois/odoo/13.0/odoo/addons/web_editor/models/ir_ui_view.py", line 25, in render
    return super(IrUiView, self).render(values=values, engine=engine, minimal_qcontext=minimal_qcontext)
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_ui_view.py", line 1182, in render
    return self.env[engine].render(self.id, qcontext)
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_qweb.py", line 58, in render
    result = super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/qweb.py", line 260, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/qweb.py", line 339, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0], encoding='unicode'), name)
odoo.addons.base.models.qweb.QWebException: invalid literal for int() with base 10: ''
Traceback (most recent call last):
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/qweb.py", line 332, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_web_webclient_bootstrap_126
  File "<template>", line 2, in body_call_content_124
  File "<template>", line 3, in set_123
  File "/home/lois/odoo/13.0/odoo/odoo/addons/base/models/ir_qweb.py", line 296, in _get_asset_nodes
    files, remains = self._get_asset_content(xmlid, options)
  File "/home/lois/odoo-addons/13.0/oca/web/web_company_color/models/ir_qweb.py", line 17, in _get_asset_content
    return ([], [asset.get_company_color_asset_node()])
  File "/home/lois/odoo-addons/13.0/oca/web/web_company_color/models/assetsbundle.py", line 16, in get_company_color_asset_node
    request.httprequest.cookies.get("cids", "").split(",")[0]
ValueError: invalid literal for int() with base 10: ''

Error to render compiling AST
ValueError: invalid literal for int() with base 10: ''
Template: web.webclient_bootstrap
Path: /t/t/t[1]/t[6]
Node: <t t-call-assets="web_company_color.company_color_assets"/> - - -
2020-04-07 12:23:57,470 24502 INFO mrp_production_request_v13 werkzeug: 127.0.0.1 - - [07/Apr/2020 12:23:57] "GET /favicon.ico HTTP/1.1" 404 - 2 0.001 0.002
2020-04-07 12:24:24,540 24502 INFO mrp_production_request_v13 werkzeug: 127.0.0.1 - - [07/Apr/2020 12:24:24] "POST /longpolling/poll HTTP/1.1" 200 - 22 0.028 50.032
2020-04-07 12:24:30,545 24502 INFO mrp_production_request_v13 werkzeug: 127.0.0.1 - - [07/Apr/2020 12:24:30] "POST /longpolling/poll HTTP/1.1" 200 - 21 0.014 50.017

```

</details>

File: web_company_color/models/assetsbundle.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+class AssetsBundleCompanyColor(AssetsBundle):
+    def get_company_color_asset_node(self):
+        """
+        Process the user active company scss and returns the node to inject
+        """
+        # For some reason now need get the active company from cookies,
+        # i don't know if exists other better way to do that
+        active_company_id = int(
```
Comment (AdriaGForgeFlow) [inline web_company_color/models/assetsbundle.py]: If there is only one company available, this piece fails as @LoisRForgeFlow mentioned above. This does the trick:
```python
try:
    active_company_id = int(
        request.httprequest.cookies.get("cids", "").split(",")[0]
    )
except Exception:
    active_company_id = False
```

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]

File: web_company_color/models/assetsbundle.py (hunk 1 above)
Comment (Tardo) [reply web_company_color/models/assetsbundle.py]: Thanks!

**Review by AdriaGForgeFlow — APPROVED** [review]

**Review by AdriaGForgeFlow — CHANGES_REQUESTED** [review]
It seems to be only working if the `--dev=xml `parameter is set in Odoo. Otherwise the company color does not update when changing companies (only updates the first time).

Peek 2020-04-08 09-21

cc ~ @JoanSForgeFlow

**Comment by Tardo** [comment]
@AdriaGForgeFlow fuuu***** I will see it later thanks :/ and perhaps split this module in two, one to allow force non-cached templates and this to the colors stuff.

**Comment by AdriaGForgeFlow** [comment]
> @AdriaGForgeFlow fuuu***** I will see it later thanks :/ and perhaps split this module in two, one to allow force non-cached templates and this to the colors stuff.

Thanks for the effort! Yeah maybe makes sense to split it in two, the one to allow force non-cached templates could be reused in other cases.

**Comment by Tardo** [comment]
Changes done! Finally i don't split the module.

**Review by AdriaGForgeFlow — APPROVED** [review]
Works like a charm now! Thanks for the work!

**Review by JoanSForgeFlow — APPROVED** [review]
Works great! Thank you!

**Pull request merged**